### PR TITLE
Cluey

### PR DESCRIPTION
**Pull request by J2thatsme**

### CODE REVIEW

File: types/comtac/Cluey/digital.schema.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    },
+    "limit": {
+      "title": "Limit",
+      "description": "Limit",
+      "type": "boolean"
+    },
+    "state": {
+      "title": "State",
+      "description": "State",
+      "type": "integer"
```
Comment (mariusgiger) [inline types/comtac/Cluey/digital.schema.json:41]: what are the possible states?

File: types/comtac/Cluey/lifecycle.schema.json [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      "type": "integer",
+      "minimum": 0,
+      "maximum": 100,
+      "title": "Battery charge",
+      "description": "The battery charge in percent.",
+      "unit": "%"
+    },
+    "batteryPowered": {
+      "title": "Battery powered",
+      "description": "Is the battery powered",
```
Comment (mariusgiger) [inline types/comtac/Cluey/lifecycle.schema.json:17]: Whether the battery is powered.

Comment (mariusgiger) [reply types/comtac/Cluey/lifecycle.schema.json:17]: I would rather formulate in passive voice than a question, but this is more stylistic

File: types/comtac/Cluey/uplink.js [hunk 3]
```diff
@@ -0,0 +1,1762 @@
...
+    fPort: port,
+    bytes,
+  });
+
+  const { device } = decoded.data.payload;
+  const { data } = decoded.data.payload;
+  const { warnings } = decoded;
+
+  // Digital inputs
+  for (let i = 0; i < 8; i++) {
```
Comment (mariusgiger) [inline types/comtac/Cluey/uplink.js:1730]: better check the lengths of data.digitalInputs

File: types/comtac/Cluey/uplink.js [hunk 4]
```diff
@@ -0,0 +1,1762 @@
...
+    digitalInputs.cyclic = digital.cot.cyclic;
+
+    digitalInputs.blocked = digital.status.blocked;
+    digitalInputs.state = digital.status.state;
+
+    const timestamp = new Date(digital.timestamp.string);
+
+    emit("sample", {
+      data: digitalInputs,
+      topic: "digital",
```
Comment (mariusgiger) [inline types/comtac/Cluey/uplink.js:1747]: does it make sense to have a separate topic per digital input?

**Review by mariusgiger — COMMENTED** [review]
looks good a few comments

File: types/comtac/Cluey/digital.schema.json (hunk 1 above)
Comment (J2thatsme) [reply types/comtac/Cluey/digital.schema.json:41]: Thats not documented

File: types/comtac/Cluey/uplink.js (hunk 4 above)
Comment (J2thatsme) [reply types/comtac/Cluey/uplink.js:1747]: As this sensor is a bridge, the topics and sorting may be need changed anyways by the client. So i think its alright for the moment.

Comment (mariusgiger) [reply types/comtac/Cluey/uplink.js:1747]: changing topic in the future is always cumbersome for people using the sensor

**Review by mariusgiger — DISMISSED** [review]

File: types/comtac/Cluey/lifecycle.schema.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+      "type": "boolean"
+    },
+    "lowSupplyVoltage": {
+      "title": "Low supply voltage",
+      "description": "Whether there is a low supply voltage",
+      "type": "boolean"
+    },
+    "timeSynced": {
+      "title": "Time synced",
+      "description": "Whether there time is synced",
```
Comment (mariusgiger) [inline types/comtac/Cluey/lifecycle.schema.json:48]: Whether the time is synced

File: types/comtac/Cluey/meta.json [hunk 6]
```diff
@@ -0,0 +1,14 @@
+{
+  "name": "LPN MS-3",
```
Comment (mariusgiger) [inline types/comtac/Cluey/meta.json:2]: Cluey?

**Review by mariusgiger — COMMENTED** [review]
Please check the device type name, and the open discussion regarding topic naming

**Review by mariusgiger — DISMISSED** [review]

**Review by mariusgiger — APPROVED** [review]

**Pull request merged**